api/lib/models/admin/index.ts

```diff
@@ -0,0 +1,35 @@
+import * as admin from 'firebase-admin';
+import { UidType } from '../../JSCommon/common-types';
+import { AuthLevel } from '../../services/auth/auth-types';
+import { IDataMapper, IDbResult } from '../../services/database';
+import { AdminDataMapperImpl } from './admin-data-mapper-impl';
+import { EmailHistory } from './types/email-history';
+import UserRecord = admin.auth.UserRecord;
+
+interface IEmail {
+  name: string;
+  substitutions: Map<string, string>;
+  emailId: string;
+}
+
+interface IAdminDataMapper extends IDataMapper<any> {
+  getEmailFromId(id: UidType): Promise<IDbResult<UserRecord>>;
+
+  sendEmails(
+    emails: IEmail[],
+    html: string,
+    senderUid: UidType,
+    subject: string,
+    fromEmail?: string,
```

Comment:
This might be because I'm unfamiliar with the language, but is there supposed to be a '?' in 'fromEmail?'

Reply:
There is. It’s an optional field. But perhaps a comment explaining that is needed.